numpyro/primitives.py

```diff
@@ -81,7 +82,12 @@ def __init__(self, fn: Optional[Callable] = None) -> None:
     def __enter__(self) -> None:
         _PYRO_STACK.append(self)
 
-    def __exit__(self, exc_type, exc_value, traceback) -> None:
+    def __exit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        exc_value: Optional[BaseException],
+        traceback: Optional[TracebackType],
```

Comment:
Taken from https://adamj.eu/tech/2021/07/04/python-type-hints-how-to-type-a-context-manager/